posts/lumen_ai_announcement/index.ipynb

```diff
@@ -0,0 +1,265 @@
+    "\n",
+    "#### Data Sources\n",
+    "\n",
+    "Let's say the company you are working for has a Snowflake database containing all business operation data. Here at [Anaconda](https://anaconda.com/) Snowflake might hold datasets of conda downloads by package, a list of our customers and their subscriptions, results of marketing campaigns and much more. To make this data available to Lumen AI we might configure a `SnowflakeSource`, [set up OAuth](https://panel.holoviz.org/how_to/authentication/configuration.html) using the existing Panel integrations and configure the OAuth provider with the Snowflake authorization server. This allows us to use the access token to authorize with the Snowflake as the user accessing the app, ensuring that Lumen AI only receives the permissions that are granted to the user accessing the app.\n",
+    "\n",
+    "```python\n",
+    "source = SnowflakeSource(\n",
+    "    account=...,\n",
+    "    authenticator='token',\n",
+    "    token=pn.state.access_token\n",
+    ")\n",
+    "\n",
+    "lmai.ExplorerUI(source).servable()\n",
+    "```\n",
+    "\n",
+    "#### Analyses\n",
+    "\n",
+    "Another avenue for customization is the ability to define custom `Analysis` classes that the LLM can invoke and that can be automatically suggested to the user based on the current context (i.e. depending on the dataset that is currently loaded). As an example, let's say the user has asked to see the table of conda downloads with detailed breakdowns per category. We can easily implement a custom `Analysis` that detects the presence of certain column(s) and suggests to the user that they can generate a report of conda downloads over the past month or year. This way we can automatically empower users to generate custom reports and analysis given the current context and also allow the agents to invoke these directly.\n",
+    "\n",
+    "```python\n",
+    "class CondaReport(Analysis):\n",
+    "    \"\"\"\n",
+    "    Summarizes conda download statistics.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    @classmethod\n",
+    "    async def applies(cls, pipeline):\n",
+    "        return 'pkg_name' in pipeline.data and 'counts' in pipeline.data\n",
+    "\n",
+    "    def __call__(self, pipeline):\n",
+    "        ...\n",
+    "        return Column(\n",
+    "            '# Conda Package Report',\n",
+    "            Markdown(summary),\n",
+    "            ...\n",
+    "        )\n",
+    "```\n",
```

Comment:
To make this example more concrete, I think you should show how `pkg_name` and `counts` is used in `__call__`, i.e. have a calculation performed, like:

```python
df = pipeline.data
total = df.query("pkg_name == 'lumen'")["counts"].sum()
summary = f"Downloads: {total}" + ...
```

Reply:
Sounds good.